lib/logstash/filters/kubernetes_metadata.rb

```diff
@@ -0,0 +1,288 @@
+# -----------------------------------------------------------------
+#
+# You would end up with an output of...
+#
+# [source, ruby]
+# -----------------------------------------------------------------
+# {
+#   "path" => "/logs/kube-logs/ssl-manager-535881469-x6hmm_storj-prod_ssl-manager-c817d2905d339677288ff73375856a066d4b4d8d45482e1f2e234428d217eb19.log",
+#   "kubernetes" => {
+#     "log_format_stdout" => "default",
+#     "pod" => "ssl-manager-535881469-x6hmm",
+#     "container_name" => "ssl-manager",
+#     "log_format_stderr" => "default",
+#     "namespace" => "storj-prod",
+#     "replication_controller" => "ssl-manager-535881469",
+#     "annotations" => {
+#       "kubernetes_io-created-by" => "{\"kind\":\"SerializedReference\",\"apiVersion\":\"v1\",\"reference\":{\"kind\":\"ReplicaSet\",\"namespace\":\"storj-prod\",\"name\":\"ssl-manager-535881469\",\"uid\":\"aca823ae-e271-11e6-99b2-42010a800002\",\"apiVersion\":\"extensions\",\"resourceVersion\":\"19491631\"}}\n"
+#     },
+#     "container_id" => "c817d2905d339677288ff73375856a066d4b4d8d45482e1f2e234428d217eb19",
+#     "labels" => {
+#       "app" => "ssl-manager",
+#       "pod-template-hash" => "535881469",
+#       "version" => "latest"
+#     }
+#   },
+#   "@timestamp" => 2017-02-13T02:33:11.702Z,
+#   "@version" => "1",
+#   "host" => "floptop.local",
+#   "message" => "{\"log\":\"10.244.1.1 - - [25/Jan/2017:14:51:38 +0000] \\\"GET / HTTP/1.1\\\" 200 0 \\\"-\\\" \\\"GoogleHC/1.0\\\" \\\"-\\\"\\n\",\"stream\":\"stdout\",\"time\":\"2017-01-25T14:51:38.65719468Z\"}",
+#   "tags" => []
+# }
+# -----------------------------------------------------------------
+#
+
+class LogStash::Filters::KubernetesMetadata < LogStash::Filters::Base
+
+  attr_accessor :lookup_cache
+
+  config_name "kubernetes_metadata"
+
+  # The source field name which contains full path to kubelet log file.
+  config :source, :validate => :string, :default => "path"
+
+  # The target field name to write event kubernetes metadata.
+  config :target, :validate => :string, :default => "kubernetes"
+
+  # Auth for hitting the Kubernetes API. This can be either basic auth or
+  # Bearer auth. It should be formated like the following
+  # [source, ruby]
+  # -----------------------------------------------------------------
+  # auth => { # Authentication for your API in the form of either basic or bearer auth
+  #   basic => {
+  #     user => "mykubeusername"
+  #     pass => "superwonderfulsecurepassword"
+  #   }
+  #
+  #   # ...or...
+  #
+  #   bearer  => {
+  #     key => "mybearerkeystring"
+  #   }
+  # }
+  # -----------------------------------------------------------------
+  config :auth, :validate => :hash, :default => {}
+
+  # Kubernetes API URL
+  config :api, :validate => :string, :default => "http://127.0.0.1:8001"
+
+  # Default log format
+  config :default_log_format, :validate => :string, :default => "default"
```

Comment:
Is the docs here enough information for a user to decide? What other log format values would there be?